### PR TITLE
Improve the entire development setup

### PR DESCRIPTION
**Pull request by LucianBuzzo**
Fixes: https://github.com/balena-io-modules/rendition/issues/811
Fixes: https://github.com/balena-io-modules/rendition/issues/997

Notable changes:
- Remove screenshot testing in favor of snapshot testing using storybook stories. We had a lot of issues with screenshot tests, some of which were flakiness, slowing down development and contributions. Although there is a benefit of having screenshot tests, we decided to optimize for development speed instead, since visual changes are not as common recently. We might introduce them again in the future only if they are stable and fast. Generating storybook preview deployments is a better alternative IMO.

- Upgrade to storybook 6. This was a big one, that simplified our tooling (no more custom webpack setups, etc.), but it required a lot of rewrite on how we write stories. One of the main benefits is the autogenerated Docs page for each component where you can interact with the component and change props on the fly. This also means we removed the `README.md` files in favor of inline comments inside the source code that gets picked up by SB 6.

- Moved everything to Typescript. Although the components code was in TS, the tests and stories weren't which meant we had 2 different development tools (eslint + balena-lint), and this just converges to a single tooling stack.

- Removed a lot of snapshot tests (and `spec.ts` files) since the storyshot testing was superior. This means we only need to write unit tests for more complex components, and the storyshot tests should cover the rest.

- Switch to named exports for all components (outside of Extra). This shouldn't be a breaking change unless you imported the components by deep imports (which you shouldn't do)

- Removed babel as it is no longer needed (we use ts-jest for the tests now)

I won't ask you to review the code, but I would ask you to just clone the branch and run it locally, and share any opinions or issues you might notice. There are some rough edges that need to be polished, and I'd prefer to tackle the non-critical ones separately (by putting them here: https://github.com/balena-io-modules/rendition/issues/1215). Also, please try to install the branch version on your project and see if there are any issues (there shouldn't be any, but as a sanity check)

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  stories: ['../src/**/story.js'],
+  addons: [
+    'storybook-readme',
+    'storybook-dark-mode',
+    '@storybook/addon-actions',
+    'storycap'
```
Comment (sradevski) [inline .storybook/main.js:7]: We don't need this anymore :)

**Review by sradevski — COMMENTED** [review]
Code-wise LGTM, but I assume it's not working yet since there are no new snapshots :)

File: src/components/Accordion/index.tsx [hunk 2]
```diff
@@ -60,6 +60,8 @@ interface InternalAccordionProps {
 }
 
 export type AccordionProps = InternalAccordionProps & RenditionSystemProps;
+
+/** [View story source](https://github.com/balena-io-modules/rendition/blob/master/src/components/Accordion/Accordion.stories.tsx) */
```
Comment (sradevski) [inline src/components/Accordion/index.tsx:64]: You can actually see the code in the storybook docs page, so this is unnecessary.

**Comment by sradevski** [comment]
@balena-ci retest

**Review by sradevski — APPROVED** [review]

**Comment by sradevski** [comment]
@balena-ci retest

**Comment by kaisoz** [comment]
@balena-ci retest

**Pull request merged**